adapters/folder/icloud.go

```diff
@@ -0,0 +1,78 @@
+package folder
+
+import (
+	"encoding/csv"
+	"errors"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/simulot/immich-go/internal/assets"
+	"github.com/simulot/immich-go/internal/gen"
+)
+
+type iCloudMeta struct {
+	albums               []assets.Album
+	originalCreationDate time.Time
+}
+
+func UseICloudAlbum(m *gen.SyncMap[string, iCloudMeta], fsys fs.FS, filename string) (string, error) {
+	file, err := fsys.Open(filename)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	albumName := strings.TrimSuffix(filepath.Base(filename), filepath.Ext(filename))
+	reader := csv.NewReader(file)
+	records, err := reader.ReadAll()
```

Comment:
Could the file to be too huge to be read in memory?

Reply:
I read that the maximum album size on icloud photos is 5000. So I would say its safe but I can also refactor to read line by line.

Reply:
Fine with me. You can read it in the memory